### PR TITLE
Update index.php

### PR DESCRIPTION
**Pull request by Yay295**
Reordered `<head>` and added basic Open Graph tags.
Changed how `?video=` works, so you no longer have to specify a file extension (and it's also case-insensitive).

### CODE REVIEW

**Comment by QuadPiece** [comment]
May I ask how extensively this is tested? I broke the server I usually run tests on :^) so it'll take a while for me to test this before merging.

**Comment by Yay295** [comment]
It's up in my test folder:
http://openings.moe/Yay295/test/test.html?video=Opening1-NekomonogatariKuro
http://openings.moe/Yay295/test/list/

That's the only video you can test with though since I don't really want to copy all of /video/*.

Watch the url when you try these:
?video=Opening1-NekomonogatariKuro.webm
?video=OPENING1-NEKOMONOGATARIKURO
?video=OpEnInG1-nEkOmOnOgAtArIkUrO

**Comment by Yay295** [comment]
Apparently not well enough. I'm fixing it.

**Comment by Yay295** [comment]
Alright, it's up in my test folder:
http://openings.moe/Yay295/test/list/
http://openings.moe/Yay295/test/

Watch the url when you try these:
?video=Opening1-NekomonogatariKuro.webm
?video=OPENING1-NEKOMONOGATARIKURO
?video=OpEnInG1-nEkOmOnOgAtArIkUrO
?video=opening1-nek


Also, these changes should allow periods to be used in file names without issue.

**Comment by QuadPiece** [comment]
Seems mostly good, however I believe the latter one shouldn't be an option. Allowing the use of shortened urls will make it hit the first match. Let's say someone links to `?video=Opening1-Acc` (for `?video=Opening1-AcchiKocchi`) - After that we'd upload some file with a different name, such as `?video=Opening1-AccelWorld`

After this, `?video=Opening1-Acc` would resolve to `Opening1-AccelWorld` instead of `Opening1-AcchiKocchi` due to it being earlier in the list when sorted alphabetically. It might also cause issues in some other cases. Such as a K-on! opening. If they for some reason get removed, their openings would instead lead to the K-on!! ones.

As well as common issues where `?video=Opening1-HunterXHunter` would resolve to the original 1999 release (even though we don't have that yet). Even if most people these days are probably looking for the 2011 version of the anime, which we have available as `?video=Opening1-HunterXHunter(2011)`. In this case. `?video=Opening1-HunterXHunter` would still lead to the original even if we added an "(1999)" at the end. This could confuse the (suprisingly many) people who aren't ware that the 2011 version is a remake/extension of the old one.

imo I think the full filename up to the extension should be required. It might be a better idea to look into an alias instead system for series with really long names. i.e a list where `?video=Opening1-DysfunctionalSystems` or `?video=Opening1-LearningToManageChaos` would redirect to the stupidly long `?video=Opening1-DysfunctionalSystemsLearningToManageChaos`

**TL;DR: I think we should require the full filename until the extension. There's going to be a lot of errors and confusion if we don't require full filenames and just assume the rest. We could instead consider some form of alias system in the future**

**Comment by Yay295** [comment]
It will only accept the full name now, except...

    ?video=Opening1-NekomonogatariKuro.AnythingAfterTheLastPeriodWillBeRecognizedAsAFileExtensionSoThisWouldBeFine

**Comment by Yay295** [comment]
Oh, also, if we had two files, `A.ext` and `A.B.ext`, and a user requested `?video=A.B`, they could get `A.ext` instead because the `.B` would be incorrectly recognized as `.ext`.

**Comment by QuadPiece** [comment]
This method seems good. However, I am unsure why. But initial video load times are now multiplied by 3.

Or to be specific. The time from resource load until video playback starts is approx 2.2x as long on an i5-3470 desktop. And 3.1x as long on an i3-4010U laptop. When simply comparing http://openings.moe/ to http://openings.moe/Yay295/test/ and also forcing caching on your test instance that is. We should look into this before we push it.

Are you able to replicate this on your end? Longer times from initial page load until video playback starts that is.

**Comment by Yay295** [comment]
How are you timing that? Just looking at the time it takes to load the files, it's nearly identical for me.

**Comment by QuadPiece** [comment]
Resource load time is 640ms on both (Measured with Firefox's tools) - But time until the video playback starts is about 1.3 on openings.moe and almost 4 seconds on your test directory (Measured with a good ol' stopwatch)

**Comment by Yay295** [comment]
I'm not seeing any difference. It took between 1 and 2 seconds to start playing for me with either version. I used
https://openings.moe/?video=Opening1-CowboyBebop.webm
and
https://openings.moe/Yay295/test/?video=Opening1-CowboyBebop
to test because the audio starts as soon as the video does so it's easier to notice.

**Comment by QuadPiece** [comment]
Very well, if you can't reproduce it, then it must be something on my end. I'll merge and deploy this. If something does pop up we can just revert it.
